SCClassLibrary/Common/Quarks/QuarksGui.sc

```diff
@@ -468,7 +473,7 @@ QuarkDetailView {
 		if(tags.size == 0 or: {
 			remoteLatest != model.git.shaForTag(tags.first);
 		}, {
-			versions.add([remoteLatest, "LATEST"]);
+			versions.add([remoteLatest, "Latest master"]);
```

Comment:
how would you feel about changing the other all-caps messages to title case too? ("DIRTY" and "INSTALLED")

Reply:
I actually wonder if "dirty" is really the right word. For newcomers, it is not understandable, and for those who know their way around, it might be intentional, so that it is counter-intuitive, too. Perhaps "uncommitted changes" may be better?

Reply:
good point. "Dirty commit" or "Uncommitted changes" are both fine with me.

Reply:
I'm not sure if "DIRTY" has a specific meaning. In the method `makeVersionTree`, there is some logic connected to this, and in line 528, there is `this.checkout(refspec)`, which in the present case results in ` Quarks.install(model.url, "DIRTY");`. 

All this seems not correct anyway, still I'm unsure if it can be touched without problems. @crucialfelix ?

Reply:
Dirty is a common source control term meaning that some of the files have been edited locally. That means that you would over write changes the user has made, which would probably piss them off.

Dirty is also used in GUI programming to mean that the user has entered some value in a form.

Or a document is dirty after you edit it but haven't yet saved

Reply:
good, thanks for your comments on this!

>That means that you would over write changes the user has made, which would probably piss them off.

Hm, yes, this is how I would have understood it - but does this apply here at all?

>Or a document is dirty after you edit it but haven't yet saved

That is closer, but is not really true in this case: the document is in fact saved, the changes are just not yet committed. But I sort of see the point.

Do you see any clearer way of putting it?

Reply:
https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html#def_dirty

it is an official git term, simple as that.

Reply:
yes, sure. So let's assume the users know that.

Reply:
By this:
> Or a document is dirty after you edit it but haven't yet saved
I mean that the word is commonly used in that context. 

It is a git term, this is a git repo. Don't invent new terminology or change it. It doesn't help users, it adds a new layer of confusion.

Reply:
> Quarks.install(model.url, "DIRTY")

DIRTY is not a refspec or branch name or tag or hash. There is no way you can use git to checkout "dirty" anything.  I'm not sure where the code above you refer to comes from

Reply:
> Hm, yes, this is how I would have understood it - but does this apply here at all?

It definitely does. If you push a button to update a quark, you would overwrite any changes you made to that quark.

When I wrote this I decided to keep it simple and just refuse to overwrite changes. If the user is knowledgeable enough to be editing code in a quark then they should use normal code tools to finish or trash what they are working on.  ie. no GUI dialogs ("overwrite changes?")

Reply:
>DIRTY is not a refspec or branch name or tag or hash. There is no way you can use git to checkout "dirty" anything. I'm not sure where the code above you refer to comes from


https://github.com/supercollider/supercollider/blob/f5ea012357f9302e0dfa5d48cf9ce9526140473f/SCClassLibrary/Common/Quarks/QuarksGui.sc#L494

https://github.com/supercollider/supercollider/blob/f5ea012357f9302e0dfa5d48cf9ce9526140473f/SCClassLibrary/Common/Quarks/QuarksGui.sc#L503

https://github.com/supercollider/supercollider/blob/f5ea012357f9302e0dfa5d48cf9ce9526140473f/SCClassLibrary/Common/Quarks/QuarksGui.sc#L523

Reply:
If it allows you to push a button and try to check out "DIRTY" then it's definitely a bug. It was written to just display, but it looks like I added a button to make an action out of it?

Reply:
Yes. I think you just conveniently reused the refspec field for a non-existing one (a bit like we use `nil` for the edge cases often). It is a mix of display and function.